lkmpg.tex

```diff
@@ -219,18 +219,95 @@ \subsection{The Simplest Module}
 \begin{code}
 obj-m += hello-1.o
 
+PWD := $(CURDIR)
+
 all:
 	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
 
 clean:
 	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
 \end{code}
 
-And finally just:
+In \verb|Makefile|, \verb|$(CURDIR)| can set to the absolute pathname of the current working directory(after all \verb|-C| options are processed, if any). See more about \verb|CURDIR| in \href{https://www.gnu.org/software/make/manual/make.html}{GNU make manual}.
+
+And finally just run \verb|make| directly.
+
 \begin{codebash}
 make
 \end{codebash}
 
+If there is no \verb|PWD := $(CURDIR)| statement in Makefile, note that using \verb|sudo make| might not compile correctly. 
+Because some environment variables can't be inherited, which is specified by the security policy. 
+The default security policy is \verb|sudoers|.
+In the \verb|sudoers| security policy, \verb|env_reset| is enabled by default, which restricts environment variables. 
+Specifically, path variables are not retained from the user environment, they are set to default values (For more information see: \href{https://www.sudo.ws/docs/man/sudoers.man/}{sudoers manual}).
+You can see the environment variable settings by:
+
+\begin{codebash}
+ubuntu@ubuntu:~$ sudo -s
+root@ubuntu:/home/ubuntu# sudo -V
+\end{codebash}
+
+Here is an simple Makefile as example to demonstrate the problem mentioned above.
+
+\begin{code}
+all:
+	echo $(PWD)
+\end{code}
+
+Then, we can use \verb|-p| flag to print out the environment variable values from the Makefile.
+
+\begin{codebash}
```

Comment:
Ditto. Annotation `codebash` should be applied to valid shell scripts or commands.